#16 [백] 회원 가입 구현

jeongbeomSeo opened the pull request:
1. Filter, Interceptor, ControllerAdvice, AOP 의 실행되는 순서 및 역할을 고려해 ControllerAdvice인 ResponseBodyAdvice에서 공통 응답인 ResponseDTO를 만들어 주는 작업 수행 => 기존의 ResponseDTO의 리스코프 치환 원칙 위배 해결 
2. 불필요한 정보를 Client Side에 보내지 않도록 ErrorResponseDTO에 담기는 필드 수정
3. SuccessResonseDTO에 불필요한 오버헤드가 늘어나지 않도록 수정
4. 에러 코드의 역할 재정립 => 클라이언트 측에서 분기점이 생길만한 상황에 에러 코드 활용
5. RestTemplate을 사용한 원격 API 호출 구축 및 예외 처리

1. Facade 패턴 적용해서 API Client 객체와 UserService 객체 의존성 떨구기
2. 회원 가입시 역할로 유저 권한 부여
3. PathVariable를 Enum 객체로 받으면서 유효성 검증
4. 404 Not Found 에러 잡아내기  
1. 회원 가입 및 이메일 중복 검사 TDD 검증 완료
2. ResponseDTO body에 null값 보내올 경우 isSuccess필드만 넘겨주도록 수정

jeongbeomSeo committed:
feat: 서버간의 트랜잭션이 빠진 회원가입 기능 구현

jeongbeomSeo committed:
feat: [백] DDD 기반 리팩토링

jeongbeomSeo committed:
chore: delete unnenessary file

jeongbeomSeo committed:
feat: 회원가입 기능 - 다른 서버와 API 통신
문제 없는 데이터의 경우 성공적으로 처리하며, 다른 서버에서 데이터 처리를 실패하는 경우에 보상 Transaction도 작동
하지만 실패한 결과를 보내오는 경우 Customizing해서 예외를 처리하지 못하고 있음

jeongbeomSeo committed:
chore: delete unnecessary file

jeongbeomSeo committed:
feat: [백] 원격 API 통신 구축 및 ResponseDTO 리팩토링 작업

1. Filter, Interceptor, ControllerAdvice, AOP 의 실행되는 순서 및 역할을 고려해 ControllerAdvice인 ResponseBodyAdvice에서 공통 응답인 ResponseDTO를 만들어 주는 작업 수행.
2. 불필요한 정보를 Client Side에 보내지 않도록 ErrorResponseDTO에 담기는 필드 수정
3. SuccessResonseDTO에 불필요한 오버헤드가 늘어나지 않도록 수정
4. 에러 코드의 역할 재정립 => 클라이언트 측에서 분기점이 생길만한 상황에 에러 코드 활용
5. RestTemplate을 사용한 원격 API 호출 구축 및 예외 처리

jeongbeomSeo committed:
feat: 회원 가입 추가 기능 추가
1. Facade 패턴 적용해서 API Client 객체와 UserService 객체 의존성 떨구기
2. 회원 가입시 역할로 유저 권한 부여
3. PathVariable를 Enum 객체로 받으면서 유효성 검증
4. 404 Not Found 에러 잡아내기

kilhyeongyeong commented on src/authservice/src/main/java/com/eventty/authservice/common/Enum/SuccessCode.java:
```diff
@@ -0,0 +1,20 @@
+    // Create
+    USER_CREATED(201);
+
+    private final int status;
```
SuccessCode에 status밖에 없는 굳이 Enum 클래스를 갖고 가시려는 이유는 차후 확장성 때문인가요?
이유가 궁금합니다..!

jeongbeomSeo replied:
아뇨 이와 같이 해두면서 ResponseEntity<SuccessResponseDTO>를 만들 때 Enum 객체를 활용하려고 남겨두었습니다 !

jeongbeomSeo replied:
상태코드는 계속해서 반복해서 사용되서 하나로 묶어둘 수 있는 이점을 가진 ENUM객체를 활용했다고 생각하시면 될 것 같습니다

jeongbeomSeo committed:
feat: 회원 가입 기능 구현 완료
1. 회원 가입 및 이메일 중복 검사 TDD 검증 완료
2. ResponseDTO body에 null값 보내올 경우 isSuccess필드만 넘겨주도록 수정

jeongbeomSeo committed:
refactoring: 도커 작동 확인 및 Spring Security설정에서 H2 접근 허용

jeongbeomSeo committed:
chore: Conflicts 해결

jeongbeomSeo committed:
Merge branch 'develop' into feature/#16
